context/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { createContext, ReactNode } from "react";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { ContextInterface } from "../types";
+import { ContextDefaultValues } from "../constants";
+import { Props } from "next/script";
+
+export const NavigationContext = createContext<ContextInterface>(ContextDefaultValues);
+
+export const NavigationProvider = ({ children }: Props) => {
+  const [offset, setOffset] = useState(1);
+  const [data, setData] = useState([]);
+  const [limit, setLimit] = useState(5);
+  const [paginationActive, setPaginationActive] = useState(false);
```

Comment:
use `is` and `has` prefixes for boolean flags - `isPaginationActive`, `setIsPaginationActive`